sklearn/metrics/_classification.py

```diff
@@ -1116,7 +1116,18 @@ def f1_score(
     The relative contribution of precision and recall to the F1 score are
     equal. The formula for the F1 score is::
 
-        F1 = 2 * (precision * recall) / (precision + recall)
+        F1 = 2 * TP / (2 * TP + FN + FP)
+
+    Where "TP" is the number of true positives, "FN" is the number of false
+    negatives, and "FP" is the number of false positives.
+
+    This is equivalent to::
+
+        F1 = 2 * precision * recall / (precision + recall)
+
+    except in the case where precision and recall are both 0. It is recommended
+    to refer to the previous definition of F1, especially for understanding the
+    behavior of the ``zero_division`` parameter.
```

Comment:
I think that we can skip the details about the case of 0.

Reply:
Are you saying that I should say that

```
F1 = 2 * TP / (2 * TP + FN + FP)
```

is equivalent to 

```
F1 = 2 * precision * recall / (precision + recall)
```

(without specifying that the latter is undefined when precision & recall are 0)?

I disagree with writing this because:

1. It is not true
2. It is very confusing and will mislead people about the purpose of the `zero_division` parameter

Reply:
I think you can just drop the comparison to `2PR/(P+R)` here and say, defined as 0 when TP=0.
For the rest, we have the user guide.